modules/shared/src/main/scala/io/constellationnetwork/currency/schema/currency.scala

```diff
@@ -79,7 +80,8 @@ object currency {
     lastMessages: Option[SortedMap[MessageType, Signed[CurrencyMessage]]],
     lastFeeTxRefs: Option[SortedMap[Address, TransactionReference]],
     lastAllowSpendRefsProof: Option[SortedMap[Address, AllowSpendReference]],
-    activeAllowSpends: Option[SortedMap[Address, List[Signed[AllowSpend]]]]
+    activeAllowSpends: Option[SortedMap[Address, List[Signed[AllowSpend]]]],
+    activeSpendTransactions: Option[SortedMap[Address, List[SpendTransaction]]]
```

Comment:
I think we need to keep them per amm metagraph, so you should wrap it by another map of addresses. Also, we would need probably the observed ordinal as for allow spends (@kpudlik?)

Reply:
Ok, I see it's a schema of currency, so this one is correct. Schema in GlobalSnapshotInfo would be different

Reply:
but, `SpendTransaction` does not have a `source` address or anything like that. What is this `Address` type in  `Option[SortedMap[Address, List[SpendTransaction]]]`?